src/components/checklist/SingleChecklistItem.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import { Container, Grid, Checkbox, Accordion, Icon, Popup, Label } from 'semantic-ui-react';
+import style from './css/SingleChecklistItem.module.css'
+import ReactMarkdown from 'react-markdown'
+
+class SingleChecklistItem extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            data: this.props.propsData,
+            priorityColor: null,
+            priorityString: null,
+            activeIndex: null,
+            iconName: 'chevron down',
+            isChecked: this.props.checkedIndex,
+            readyMDFile: null
+        };
+    }
+
+    componentWillMount() {
+        let priorityColor, priorityString;
+
+        if (this.state.data.priority === 0) {
```

Comment:
create const array with colors and labels instead conditions and don't set it in lifecycle hook at all 
example: 
`const priorities = [
  { color: 'green', label: 'Low'},
  { color: 'yellow', label: 'Medium'},
  { color: 'red', label: 'High'},
]`
in render: 
`<Icon name="circle" size="tiny" 
color={priorities[data.priority].color} 
className={style.priorityCircle} />`